src/taxonomy/delete-dialog/index.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import {
+  ActionRow,
+  Button,
+  Form,
+  ModalDialog,
+  Icon,
+} from '@edx/paragon';
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Warning } from '@edx/paragon/icons';
+import messages from './messages';
+
+const DeleteDialog = ({
+  taxonomyName,
+  tagsCount,
+  isOpen,
+  onClose,
+  onDelete,
+}) => {
+  const intl = useIntl();
+  const [deleteButtonDisabled, setDeleteButtonDisabled] = useState(true);
+  const deleteLabel = intl.formatMessage(messages.deleteDialogConfirmDeleteLabel);
+
+  const handleInputChange = (event) => {
+    if (event.target.value === deleteLabel) {
+      setDeleteButtonDisabled(false);
+    } else {
+      setDeleteButtonDisabled(true);
+    }
+  };
+
+  const onClickDelete = () => {
+    onClose();
+    onDelete();
+  };
```

Comment:
```suggestion
  const handleInputChange = React.useCallback((event) => {
    setDeleteButtonDisabled(event.target.value !== deleteLabel);
  });

  const onClickDelete = React.useCallback(() => {
    onClose();
    onDelete();
  }, [onClose, onDelete]);
```

If you don't wrap these handlers in `useCallback`, they change and become new functions every time, which will cause child components to re-render needlessly.

Reply:
Updated here: https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/fd18b0fb4446f4af7450b79fd7f9c634c1a84125